packages/shorebird_cli/lib/src/commands/patch/macos_patcher.dart

```diff
@@ -0,0 +1,370 @@
+        buildProgress.fail('Failed to build macos app');
+        logger.err(error.message);
+        rethrow;
+      }
+
+      try {
+        if (splitDebugInfoPath != null) {
+          Directory(splitDebugInfoPath!).createSync(recursive: true);
+        }
+        await artifactBuilder.buildElfAotSnapshot(
+          appDillPath: ipaBuildResult.kernelFile.path,
+          outFilePath: _aotOutputPath,
+          // TODO
+          // additionalArgs: splitDebugInfoArgs(splitDebugInfoPath),
+        );
+      } catch (error) {
+        buildProgress.fail('$error');
+        rethrow;
+      }
+
+      // Copy the kernel file to the build directory so that it can be used
+      // to generate a patch.
+      ipaBuildResult.kernelFile.copySync(_appDillCopyPath);
+
+      buildProgress.complete();
+    } catch (_) {
+      throw ProcessExit(ExitCode.software.code);
+    }
+
+    final appPath = artifactManager.getMacOSAppDirectory()!.path;
+    final tempDir = await Directory.systemTemp.createTemp();
+    final zippedApp = File(p.join(tempDir.path, '${p.basename(appPath)}.zip'));
+    // FIXME: using ditto here because zipToTempFile is not properly capturing
+    // the app folder structure (the top folder after zipping is Content,
+    // instead of the MyApp.app directory).
+    // package:archive also seems to be having some trouble unzipping .app files
+    //
+    // final zippedApp = await Directory(appPath).zipToTempFile();
+    await Process.run('ditto', [
+      '-c',
+      '-k',
+      '--sequesterRsrc',
+      '--keepParent',
+      appPath,
+      zippedApp.path,
+    ]);
+    print('appPath is $appPath');
+    return zippedApp;
+  }
+
+  @override
+  Future<Map<Arch, PatchArtifactBundle>> createPatchArtifacts({
+    required String appId,
+    required int releaseId,
+    required File releaseArtifact,
+    File? supplementArtifact,
+  }) async {
+    // Verify that we have built a patch .app
+    if (artifactManager.getMacOSAppDirectory()?.path == null) {
+      logger.err('Unable to find .app directory');
+      throw ProcessExit(ExitCode.software.code);
+    }
+
+    final unzipProgress = logger.progress('Extracting release artifact');
+    final tempDir = Directory.systemTemp.createTempSync();
+    await artifactManager.extractZip(
+      zipFile: releaseArtifact,
+      outputDirectory: tempDir,
+    );
+    final releaseAppPath = tempDir.path;
+
+    unzipProgress.complete();
+    final appDirectory = artifactManager.getMacosAppDirectory(
+      parentDirectory: Directory(releaseAppPath),
+    );
+    if (appDirectory == null) {
+      logger.err('Unable to find release artifact .app directory');
+      throw ProcessExit(ExitCode.software.code);
+    }
+    final releaseArtifactFile = File(
+      p.join(
+        appDirectory.path,
+        'Contents',
+        'Frameworks',
+        'App.framework',
+        'App',
+      ),
+    );
+
+    final useLinker = AotTools.usesLinker(shorebirdEnv.flutterRevision);
```

Comment:
need to unzip and copy the App.ct.link and App.class_table.json from supplementArtifact here just like in ios_patcher.dart

Reply:
That did it! Thanks